File: packages/android_intent_plus/android/src/main/java/dev/fluttercommunity/plus/androidintent/Bundle/PutClasses/PutStringArrayList.java
package dev.fluttercommunity.plus.androidintent.Bundle.PutClasses;

import android.os.Bundle;
import dev.fluttercommunity.plus.androidintent.Bundle.Constants;
import dev.fluttercommunity.plus.androidintent.Bundle.PutClasses.base.PutBase;
import java.util.ArrayList;
import java.util.List;
import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

public class PutStringArrayList extends PutBase<List<String>> {

  public static final String JAVA_CLASS = "PutStringArrayList";

  public PutStringArrayList(String key, List<String> value) {
    super(key, JAVA_CLASS, value);
  }

  public static void convert(Bundle bundle, PutStringArrayList putStringArrayList) {
    bundle.putStringArrayList(putStringArrayList.key, new ArrayList<>(putStringArrayList.value));
  }

  public static PutStringArrayList fromJson(JSONObject jsonObject) throws JSONException {
    final ArrayList<String> arrayList = new ArrayList<>();
    final JSONArray jsonArray = jsonObject.getJSONArray(Constants.VALUE);
    for (int i = 0; i < jsonArray.length(); i++) {
      arrayList.add(jsonArray.getString(i));
    }
    return new PutStringArrayList(jsonObject.getString(Constants.KEY), arrayList);
  }

  @Override
  public JSONObject toJson() throws JSONException {
    JSONObject jsonObject = new JSONObject();
    jsonObject.put("key", key);
    jsonObject.put("javaClass", JAVA_CLASS);
    jsonObject.put("value", new JSONArray(value));
    return jsonObject;
  }
}

File: packages/android_intent_plus/lib/put_classes/put_string_array_list.dart
import 'package:android_intent_plus/put_classes/base/put_base.dart';
import 'package:json_annotation/json_annotation.dart';

part 'put_string_array_list.g.dart';

@JsonSerializable()
class PutStringArrayList extends PutBase {
  factory PutStringArrayList.create({
    required String key,
    required List<String> values,
  }) =>
      PutStringArrayList(
        key: key,
        javaClass: 'PutStringArrayList',
        values: values,
      );

  PutStringArrayList({
    required String key,
    required String javaClass,
    required this.values,
  }) : super(
          key: key,
          javaClass: javaClass,
        );

  final List<String> values;

  factory PutStringArrayList.fromJson(Map<String, dynamic> json) =>
      _$PutStringArrayListFromJson(json);

  @override
  Map<String, dynamic> toJson() => _$PutStringArrayListToJson(this);
}
